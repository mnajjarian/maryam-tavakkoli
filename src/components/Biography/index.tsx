import React, { Fragment, useContext } from "react";
import Section from "../Section";
import Header from "../Header";
import Footer from "../Footer";
import { DataContext } from "../../contexts/dataContext";
import { AuthContext } from "../../contexts/authContext";
const baseUrl = "http://localhost:3001/";

interface Profile {
  name: string;
  image: string;
  biography: string;
}
const Biography = () => {
  const {
    data: { users }
  } = useContext(DataContext);
  const { authState } = useContext(AuthContext);

  if (!users.length || !authState) {
    return <div></div>;
  }
  const user = users.filter((user: any) => user._id === authState.id)[0];
  const { firstName, lastName, email, bio, imageUrl } = user;
  const fullname = firstName + " " + user.lastName;

  return (
    <Fragment>
      <Header />
      <Section
        imgUrl={imageUrl}
        title={fullname}
        href="/about"
        btnText="read more"
        flexDirect="row"
        borderRadius="50%"
        text={bio}
      />
      <Footer />
    </Fragment>
  );
};

export default Biography;
