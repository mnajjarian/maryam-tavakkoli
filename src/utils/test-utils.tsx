import React from "react";
import { render } from "@testing-library/react";
import { AuthProvider } from "../contexts/authContext";
import { DataProvider } from "../contexts/dataContext";
import { ReactNode } from "react";
import { BrowserRouter, Router } from "react-router-dom";
import { createBrowserHistory } from "history";

const history = createBrowserHistory();
interface Props {
  children: ReactNode;
}
const AllTheProviders = (props: Props) => {
  return (
    <AuthProvider>
      <DataProvider>
        <BrowserRouter>
          <Router history={history}>{props.children}</Router>
        </BrowserRouter>
      </DataProvider>
    </AuthProvider>
  );
};

const customRender = (ui: any, options: any) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";

export { customRender as render };
