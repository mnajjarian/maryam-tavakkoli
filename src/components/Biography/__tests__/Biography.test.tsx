import React from "react";
import Biography from "..";
import { render, rerender } from "test-utils";
import { async } from "q";

test("renders Biography without crashing", async () => {
  const component = render(<Biography />);
  expect(component.container).toMatchSnapshot();
});
