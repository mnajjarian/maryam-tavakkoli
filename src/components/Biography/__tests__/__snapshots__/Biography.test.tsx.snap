// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Biography without crashing 1`] = `
<div>
  <div />
</div>
`;
